build.sbt

```diff
@@ -706,6 +707,23 @@ def beamRunnerSettings: Seq[Setting[_]] = Seq(
   libraryDependencies ++= beamRunnersEval.value
 )
 
+def vectoredReadSettings: Seq[Setting[_]] = sys.props
+  .get("vectoredReadsEnabled")
+  .map(_.toBoolean)
+  .collect { case true =>
+    Seq(
+      dependencyOverrides ++= Seq(
+        "com.google.cloud.bigdataoss" % "gcs-connector" % bigdataoss3Version,
+        "com.google.cloud.bigdataoss" % "gcsio" % bigdataoss3Version,
+        "com.google.cloud.bigdataoss" % "util-hadoop" % bigdataoss3Version,
+        "org.apache.hadoop" % "hadoop-common" % "3.3.6",
+        "org.apache.hadoop" % "hadoop-auth" % "3.3.6",
+        "org.apache.hadoop" % "hadoop-client" % "3.3.6"
```

Comment:
nit: maybe move these hardcoded versions to some variable

Reply:
wait I just realized we already have `hadoopVersion` variable set to `3.4.1` which is used for `org.apache.hadoop:hadoop-common` and `org.apache.hadoop:hadoop-client`. Will this be an issue? can we just bring everything down to `3.3.6` or does this cause issue with beam's hadoop version?

Reply:
I was on the fence about that a bit!! Ultimately I left the default version as 3.4 since it does contain some (mostly) transitive security fixes.